### PR TITLE
add turkish and chinese translations

### PR DESCRIPTION
**Pull request by Anarchid**
image

Turkish translation is as complete as current ZK i18n system allows (which is sorely lacking, this shit needs to be everywhere).

Chinese is around 80% complete. For Windows users, using the Chinese translation requires generating the fontconfig cache, which Spring can do with some obscure CLI param. When/if this gets merged and goes into the release, this should be mentioned.

### CODE REVIEW

**Comment by Anarchid** [comment]
Marinating this until tomorrow if anyone wants to review.

**Comment by sprunk** [comment]
How does Chinese look without a generated font config? If it's broken it's probably best to merge just türkish and wait until wrapper runs spring with the required CLI param. 

Does the param need to be invoked on every startup, or just once and the result is cached? If it's cached but I invoke the param again anyway, do I have to regenerate it from scratch? How expensive is the regeneration? If the answers are not "once, no, cheap" then using it might require some wrapper logic to avoid the extra cost of doing this when not needed.

**Comment by Anarchid** [comment]
> If it's broken it's probably best to merge just türkish and wait until wrapper runs spring with the required CLI param.

It looks like squares.

> Does the param need to be invoked on every startup, or just once and the result is cached? 

Just once. It takes a couple minutes during Spring boot, which is black screen kinds of scary, which is why it got hidden behind the CLI param instead of being always on.

Wrapper logic would be conceptually simple, but would also need a lobby-side language selector and a button to press. This, and further expanding the scope of i18n (into lobby, foremost) is a separate concern i think.

**Comment by Anarchid** [comment]
IIRC both `Spring.Start` and `Spring.Restart` support command-line argument passing, so this could completely bypass wrapper and be done purely in Chobby lua code.

**Comment by sprunk** [comment]
Alright, then IMO in the language list file:

```lua
local langs = {
    ...
    { lang = 'tr', flag = 'tr', name = 'Türkçe' },
}

if Platform.osFamily ~= "Windows" then
    -- börkçe by default, needs to run the engine with some CLI param
    langs[#langs + 1] = { lang = 'zh', flag = 'cn', name = '中文' }
end
```

After the wrapper part is done, that check can be removed.

**Comment by Anarchid** [comment]
Ah! Yes, that's better!

File: LuaUI/Configs/lang/interface.tr.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"space_click_show_stats" : "Boşluk+Tıkla: Ünite bilgisini göster",
+	"hold_space_for_help" : "Boşluğa basılı tutarak yardım et",
+	"fac_modifier_keys_instruction" : "Shift+Tık: 5x kez\nCtrl++Tık: 10x kez\nAlt+Tık: Üniteleri kuyruğun önüne ekle\n Sağ tık: üniteyi kuyruktan kaldır",
+	"evac_ready" : "Tahliye Hazır",
+	"evac_charging" : "Tahliye Bekleniyor",
+	"evac_no_wormhole" : "Solucan deliği yok",
+	"evac_wormhole_destroyed" : "Solucan deliği yokedildi",
+	"evac_nothing_to_evac" : "Tahliye edecek birşey yok",
+	"evac_broken" : "Tahliye bozuldu",
+	"planet" : "Gezegen %{gezegen}",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: `%{planet}`

File: LuaUI/Configs/lang/common.tr.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"ready" : "Hazır",
+	"reloading" : "Yeniden yükleniyor",
+	"select" : "Seç",
+	"deselect" : "Seçimi kaldır",
+	"select_type" : "Seçme tipi",
+	"deselect_type" : "Seçim kaldırma tipi",
+	"select_all" : "Hepsini seç",
+	"select_and_go_to" : "Seç ve şuraya git",
+	"go_to" : "Şuraya git",
+	"x%_done" : "%{x}%%Tamam",
```
Comment (sprunk) [inline LuaUI/Configs/lang/common.tr.json]: Probably `%{x}%% Tamam` (with space)?

File: LuaUI/Configs/lang/common.tr.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	"tidal_income" : "Gelgit geliri",
+	"od_payback" : "Aşırı yükleme geliri",
+	"base_payback" : "Ana Gelir",
+	"never" : "Asla",
+	"selected_units" : "Seçili Üniteler",
+	"unknown" : "bilinmeyen",
+	"buildpower" : "İnşa Gücü",
+	"burst_damage" : "Patlama Hasarı",
+	"burst" : "Patlama",
+	"debris" : "Enkaz",
```
Comment (sprunk) [inline LuaUI/Configs/lang/common.tr.json]: `debris` and `wreckage` should use different names.

File: LuaUI/Configs/lang/healthbars.tr.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "paralyze": "Felç",
+    "disarm": "Silahsız bırak",
+    "capture": "Ele gerçir",
+    "capture_reload": "Yeniden yakala",
+    "water_tank": "Su tankı",
+    "teleport": "Işınlan",
+    "ability": "Yetenek",
+    "reload": "Yenile",
+    "slow": "Yavaş",
+    "goo": "goo",
```
Comment (sprunk) [inline LuaUI/Configs/lang/healthbars.tr.json]: `goo` missed?

File: LuaUI/Configs/lang/interface.tr.json [hunk 5]
```diff
@@ -0,0 +1,106 @@
+{
+	"idle_cons" : {
+		"one" : "%{sayı} tane boşta mühendisin var.",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: Keep as `%{count}`

File: LuaUI/Configs/lang/interface.tr.json [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	"pw_instruction_ip_attacker" : "Komuta Merkezinizi sağlam olarak kazanırsanız, %{att_kept_cc} etki kazanacaksınız (dropships % % {base} base +% {dropships} - savunmadan %{defence}.\n\nKomuta Merkezinizi, eğer onsuz kazanırsanız, yalnızca %{att_lost_cc} etkisi kazanacağınız için koruyun.\n\nEğer kaybediyorsanız, düşman Komuta Merkezini yoketmeye çalışın, yok ettiğinizde mağlubiyette %{def_lost_cc} etkisi olur.\n\nDüşman Komuta Merkezini tahrip etmeden mağlup olursanız, %{def_kept_cc} etkisi kazanırsınız.",
+	"pw_instruction_ip_spec" : "% {defender} zaferi,% {defender} Komuta Merkezi hayatta kalırsa,% {def_er} için% {def_kept_cc} etkisiyle sonuçlanır, aksi halde% {def_lost_cc}.\n\n%{attacker} zaferi, Komuta Merkezi tahrip edildiyse %{att_lost_cc} etkisine neden olur, bunun yerine %{att_kept_cc} (%{base} base + %{dropships} dropships - %{defence} savunma).",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: Broken tags (should be no space between `%{`)

File: LuaUI/Configs/lang/interface.tr.json [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+	"pw_instruction_who_neutral_spec" : "Tarafsız gezegen %{planet} %{attacker} tarafından istila edildi, kim %{ip} %{needed} etkiye sahipse fethedecek.",
+	"pw_instruction_ip_defender" : "Komuta Merkezinizi sağlam olarak kazanırsanız,%{attacker} %{def_kept_cc} etki kazanacaktır.\n\nKomuta Merkezinizi koruyun, eğer onsuz kazanırsanız, düşman %{def_lost_cc} etkisine sahip olacak\n\nKaybederseniz, %{saldırgan} %{att_kept_cc} etki kazanacaktır (dropships %{base} base +% {dropships} - savunmadan %{defence}.\n\nEğer kaybediyorsanız, düşman Komuta Merkezini öldürmeye çalışın, zira sadece %{att_lost_cc} zaferden etkilenecektir.",
+	"pw_instruction_ip_attacker" : "Komuta Merkezinizi sağlam olarak kazanırsanız, %{att_kept_cc} etki kazanacaksınız (dropships % % {base} base +% {dropships} - savunmadan %{defence}.\n\nKomuta Merkezinizi, eğer onsuz kazanırsanız, yalnızca %{att_lost_cc} etkisi kazanacağınız için koruyun.\n\nEğer kaybediyorsanız, düşman Komuta Merkezini yoketmeye çalışın, yok ettiğinizde mağlubiyette %{def_lost_cc} etkisi olur.\n\nDüşman Komuta Merkezini tahrip etmeden mağlup olursanız, %{def_kept_cc} etkisi kazanırsınız.",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: Duplicated and spaced `%` at "dropships"

File: LuaUI/Configs/lang/interface.tr.json [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+	"cheatsheet_lab_top" : "Hile Sayfası",
+	"cheatsheet_chb_godmode" : "Çevresel özellik kontrolü",
+	"cheatsheet_chb_godmode_desc" : "Seyircilerin tüm takımları kontrol etmesine izin verir",
+	"cheatsheet_chb_nocost" : "Hızlı İnşa",
+	"cheatsheet_chb_nocost_desc" : "İnşaat ücretsiz hale gelir ve anında tamamlanır.",
+	"cheatsheet_chb_globallos" : "Çevresel vizyon",
+	"cheatsheet_chb_globallos_desc" : "Tüm harita üzerinde görüş hattını al.",
+	"cheatsheet_chb_cheat" : "Hileleri etkinleştir",
+	"cheatsheet_chb_cheat_desc" : "Hileleri değiştirmeye izin ver. Hile kontrolü devre dışı bırakılsa bile hilelerin etkin olduğunu unutmayın.",
+	"cheatsheet_count_slider" : "Sayı: %%.den",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: Keep this `%%.0f`

File: LuaUI/Configs/lang/interface.tr.json [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+	"cheatsheet_teampick_desc" : "Üniteleri kim alıyor?",
+	"cheatsheet_pick_category" : "Üniye kategorisini seçin.",
+	"cheatsheet_pick_unit" : "Belirli bir birim seçin.",
+	"cheatsheet_no_category" : "<choose category>",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: untranslated

File: LuaUI/Configs/lang/resbars.tr.json [hunk 10]
```diff
@@ -0,0 +1,36 @@
+{
+	"local_metal_economy" : "Yerel Metal Ekonomisi",
+	"local_energy_economy" : "Yerel Enerji Ekonomisi",
+	"team_metal_economy" : "Takım Metal Ekonomisi",
+	"team_energy_economy" : "Takım Enerji Ekonomisi",
+	"resbar_base_extraction" : "Ana ihraç",
+	"resbar_overdrive" : "Aşırı Yükleme",
+	"resbar_overdrive_efficiency" : "AşırıYükleme Etkisi",
```
Comment (sprunk) [inline LuaUI/Configs/lang/resbars.tr.json]: Judging by the line above there's a missing space

File: LuaUI/Configs/lang/resbars.tr.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+	"resbar_overdrive" : "Aşırı Yükleme",
+	"resbar_overdrive_efficiency" : "AşırıYükleme Etkisi",
+	"resbar_reclaim" : "Yeniden al",
+	"resbar_cons" : "Mühendisler",
+	"resbar_sharing" : "Paylaşılıyor",
+	"resbar_construction" : "İnşaat",
+	"resbar_reserve" : "Kaynak",
+	"resbar_stored" : "Depolandı",
+	"resbar_no_storage" : "Depo Yok",
+	"resbar_inc" : "Inc",
```
Comment (sprunk) [inline LuaUI/Configs/lang/resbars.tr.json]: Possibly untranslated?

File: LuaUI/Configs/lang/campaign_units.tr.json [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    "comm_campaign_odin": {
+        "name": "Odin",
+        "helptext": "Valhalanın liderleri, savaşın ihtişamı için yaşayan, inşa edilmiş ve yetiştirilmiş bir savaşçı. Son derece acımasız bir düşman."
+    },
+    "comm_campaign_isonade": {
+        "name": "Lord Isonade",
+        "helptext": "Ubermenschen lerden biri Dynasty soylularındandır,  hırslı savaş lordu kendini galaksinin efendisi olarak tanımlıyor. Canavar olarak korkunç mutasyona uğramış."
+    },
+    "comm_campaign_legion": {
+        "name": "Legate Fidus",
```
Comment (sprunk) [inline LuaUI/Configs/lang/campaign_units.tr.json]: `Legate` is a title which is not natively English so likely can be translated.

File: LuaUI/Configs/lang/campaign_units.tr.json [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        "helptext": "Kraliyet öncü lefyonu komutanı, imparatorun iradesinin uygulayıcıları. Sadık ve güvenilir bir asker."
+    },
+    "comm_campaign_praetorian": {
+        "name": "Scipio Astra",
+        "helptext": "İmparatorluk paladinleri seçkin Praetorian Guard'ın valisi. İmparator tarikatının fanatik bir tutkusuydu, kafirleri büyük ateş gücü ile temizliyordu."
```
Comment (sprunk) [inline LuaUI/Configs/lang/campaign_units.tr.json]: `Praetorian Guard` can probably be translated as well? https://tr.wikipedia.org/wiki/Praetorian_muhafızlar

File: LuaUI/Configs/lang/units.tr.json [hunk 14]
```diff
@@ -0,0 +1,1172 @@
...
+    },
+    "striderbantha": {
+        "name": "Paladin",
+        "description": "Menzilli Destek Strideri",
+        "helptext": "Paladin, özellikle kırılmaz bir savunma hattına daha sert bir çözümdür, takyon projektörü ve duraksama nişanları için EMD füzeleri, genel amaçlı savaş için yıldırım eli topları ve çok sayıda zırh bulunur. Ancak dikkatli olun, çünkü havaya karşı savunmasızdır ve kendi başına etkin bir şekilde kullanılamaz."
+    },
+    "shipcarrier": {
+        "name": "Resif",
+        "description": "Uçak Gemisi (Bombardıman), 5 m/s de bozucu roket stoklar",
+
```
Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `AirAlani` in the helptext but `HavaAlani` in the name.

Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Recon` in the streets, `Muhalefet` in the sheets

Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: Helptext says `Merlin` but name says `Bogdan`.

Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Şovalye Komutan`?

Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Memur` was the jumpy con, this one is `Efsuncu`.

Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Pyro` vs `Ateş`

File: LuaUI/Configs/lang/units.tr.json [hunk 15]
```diff
@@ -0,0 +1,1172 @@
...
+    },
+    "striderscorpion": {
+        "name": "Akrep",
+        "description": "Görünmez Sızma Strideri",
+        "helptext": "Akrep, yıldırım iğnesiyle düşmanları felç eder ve daha sonra onları parçacık demeti pençeleri ile çiğniyor. Sert kabarması, düşman ateşinin çoğunu ortadan kaldırır ve Akrep'i şeffaf hale getirebilir."
+    },
+    "vehcapture": {
+        "name": "Sahipçi",
+        "description": "Elegeçirme Arabası",
+        "helptext": "Dominatrix, düşmanlarını eski müttefiklerine karşı çevirerek ele geçirir. Birden fazla Dominatrix, tek bir hedefi daha hızlı yakalayabilir, ancak yalnızca biri denetleyici olur. Birimin kontrolünü aldıktan sonra Dominatrix 12 saniye boyunca ateş edemez. Bir Dominatrix tahrip olursa, tüm birimler serbest kalır."
```
Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Dominatrix` -> `Sahipçi`

File: LuaUI/Configs/lang/units.tr.json [hunk 16]
```diff
@@ -0,0 +1,1172 @@
...
+    },
+    "gunshipassault": {
+        "name": "Revenant",
+        "description": "Ağır İsyan/Taarruz Kopter",
+        "helptext": "Revenant bir güdümsüz roket fırlatır. Yüksek HP'ye ve çok büyük bir hasar çıktısına sahip, ancak özellikle hareketli hedeflere karşı pahalı ve yanlış. Baz yapıları, yavaş hareket eden veya sabit birimler kümeleri üstlenmek için en iyisidir. Revenant, alınan hasarı en aza indirmek için voleybollar arasında ayrılmalıdır."
+    },
+    "tankriot": {
+        "name": "Zalim",
+        "description": "Ağır İsyan Destek Tankı",
+        "helptext": "Bir isyan platformu için oldukça hareketli olan Ogre, sınırlı bir menzile sahip olmalarına rağmen, hem hafif birimleri hem de uçakları harap eden ikiz yüksek hızlı parçalanma füzelerini bir araya getiriyor. Diğer isyan birimleri gibi, Ogre de çoğu silahlı avcıya karşı kendi ayak tutabilecek menzile ve hıza sahip değil. Füze araziyi düzlemekte oldukça etkilidir, bu nedenle Kaynakçıların ulaşamadığı duvarları yıkmada özellikle yararlıdır."
```
Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: `Ogre` -> `Zalim`

**Comment by Aquanim** [comment]
I am unsure how likely the original contributor is to look at this PR.

**Comment by Anarchid** [comment]
I did get in touch via Discord. Some of this stuff has been fixed, i just need to sync the PR with transifex.

**Comment by Anarchid** [comment]
I've fixed the things in the turkish translation that TurkeyFighter missed, and updated the turkish language files from transifex.

File: LuaUI/Configs/lang/campaign_units.tr.json [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    "comm_campaign_biovizier": {
+        "name": "Biovizier",
+        "helptext": "YZ Dynasty'nin genetik kubbelerinin ve usta genetikçinin bekçisi. Soğuk ve hesaplıyor"
+    },
+    "comm_campaign_odin": {
+        "name": "Odin",
+        "helptext": "Valhalanın liderleri, savaşın ihtişamı için yaşayan, inşa edilmiş ve yetiştirilmiş bir savaşçı. Son derece acımasız bir düşman."
+    },
+    "comm_campaign_isonade": {
+        "name": "Lord Isonade",
```
Comment (sprunk) [inline LuaUI/Configs/lang/campaign_units.tr.json]: `dynasty`, `lord`, `legate`, `praetorian guard` and `vizier` sound like translatable words.

File: LuaUI/Configs/lang/interface.tr.json [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+	"pw_instruction_who_neutral_spec" : "Tarafsız gezegen %{planet} %{attacker} tarafından istila edildi, kim %{ip} %{needed} etkiye sahipse fethedecek.",
+	"pw_instruction_ip_defender" : "Komuta Merkezinizi sağlam olarak kazanırsanız, %{attacker} %{def_kept_cc} etki kazanacaktır.\n\nKomuta Merkezinizi koruyun, eğer onsuz kazanırsanız, düşman %{def_lost_cc} etkisine sahip olacak.\n\nKaybederseniz, %{attacker} %{att_kept_cc} etki kazanacaktır (%{base} base + dropships {dropships} - savunmadan %{defense}).\n\nEğer kaybediyorsanız, düşman Komuta Merkezini öldürmeye çalışın, zira sadece %{att_lost_cc} zaferden etkilenecektir.",
+	"pw_instruction_ip_attacker" : "Komuta Merkezinizi sağlam olarak kazanırsanız, %{att_kept_cc} etki kazanacaksınız (base %{base} + dropships %{dropships} - savunmadan %{defence}).\n\nKomuta Merkezinizi, eğer onsuz kazanırsanız, yalnızca %{att_lost_cc} etkisi kazanacağınız için koruyun.\n\nEğer kaybediyorsanız, düşman Komuta Merkezini yoketmeye çalışın, yok ettiğinizde mağlubiyette %{def_lost_cc} etkisi olur.\n\nDüşman Komuta Merkezini tahrip etmeden mağlup olursanız, %{def_kept_cc} etkisi kazanırsınız.",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: `%{defense}` (with an `s`)

File: LuaUI/Configs/lang/healthbars.tr.json (hunk 4 above)
Comment (sprunk) [reply LuaUI/Configs/lang/healthbars.tr.json]: If there is no equivalent word, `reproduction` could be used as the source word.

File: LuaUI/Configs/lang/units.tr.json [hunk 19]
```diff
@@ -0,0 +1,1172 @@
...
+    },
+    "tankriot": {
+        "name": "Zalim",
+        "description": "Ağır İsyan Destek Tankı",
+        "helptext": "Bir isyan platformu için oldukça hareketli olan Zalim, sınırlı bir menzile sahip olmalarına rağmen, hem hafif birimleri hem de uçakları harap eden ikiz yüksek hızlı parçalanma füzelerini bir araya getiriyor. Diğer isyan birimleri gibi, Zalim de çoğu silahlı avcıya karşı kendi ayak tutabilecek menzile ve hıza sahip değil. Füze araziyi düzlemekte oldukça etkilidir, bu nedenle Kaynakçıların ulaşamadığı duvarları yıkmada özellikle yararlıdır."
+    },
+    "staticantinuke": {
+        "name": "Antinükleer",
+        "description": "Stratejik Nükleer Engelleme Sistemi",
+        "helptext": "Antinukleer, kapsama alanı dahilindeki düşman nükleer ROKET'leri otomatik olarak durdurur."
```
Comment (sprunk) [inline LuaUI/Configs/lang/units.tr.json]: Why is `ROKET` all-caps?

File: LuaUI/Configs/lang/campaign_units.tr.json (hunk 17 above)
Comment (Anarchid) [reply LuaUI/Configs/lang/campaign_units.tr.json]: This goes beyond repair and editing and into actual translating, so i'll leave that to whoever wants to do that on transifex.

**Comment by Anarchid** [comment]
Seems good now? World, my finger is on the button.

File: LuaUI/Configs/lang/interface.tr.json [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+	"pw_instruction_ip_attacker" : "Komuta Merkezinizi sağlam olarak kazanırsanız, %{att_kept_cc} etki kazanacaksınız (base %{base} + dropships %{dropships} - savunmadan %{defense}).\n\nKomuta Merkezinizi, eğer onsuz kazanırsanız, yalnızca %{att_lost_cc} etkisi kazanacağınız için koruyun.\n\nEğer kaybediyorsanız, düşman Komuta Merkezini yoketmeye çalışın, yok ettiğinizde mağlubiyette %{def_lost_cc} etkisi olur.\n\nDüşman Komuta Merkezini tahrip etmeden mağlup olursanız, %{def_kept_cc} etkisi kazanırsınız.",
+	"pw_instruction_ip_spec" : "%{defender} zaferi, %{defender} Komuta Merkezi hayatta kalırsa, %{attacker} için %{def_kept_cc} etkisiyle sonuçlanır, aksi halde %{def_lost_cc}.\n\n%{attacker} zaferi, Komuta Merkezi tahrip edildiyse %{att_lost_cc} etkisine neden olur, bunun yerine %{att_kept_cc} (%{base} base + %{dropships} dropships - %{defence} savunma).",
```
Comment (sprunk) [inline LuaUI/Configs/lang/interface.tr.json]: Still has `defence` instead of `defense`.

**Comment by Anarchid** [comment]
Fixed that `defence` again, also finally found where transifex keeps the search by translation string feature.

**Pull request merged**